libqtile/widget/volume.py

```diff
@@ -178,27 +180,29 @@ def setup_images(self):
                 self.length = img.width + self.actual_padding * 2
             self.surfaces[name] = img.pattern
 
+    
     def get_volume(self):
         try:
             get_volume_cmd = self.create_amixer_command("sget", self.channel)
 
             if self.get_volume_command:
                 get_volume_cmd = self.get_volume_command
 
-            mixer_out = self.call_process(get_volume_cmd)
+            self.mixer_out = self.call_process(get_volume_cmd)
         except subprocess.CalledProcessError:
             return -1
 
-        if "[off]" in mixer_out:
-            return -1
+        volgroups = re.compile(r"\[(\d?\d?\d?)%\]").search(self.mixer_out)
```

Comment:
Not sure why you changed this. You don't want to `re.compile` this every time `get_volume` is called. This has already been compiled earlier in the code.

```suggestion
        volgroups = re_vol.search(self.mixer_out)
```

Reply:
yes, I forgot to change that sorry. I used it when I was experimenting with overriding the function, but forgot to change it after copying it.